test/shape_transform_descriptor.cpp

```diff
@@ -0,0 +1,389 @@
+                         make_op("transpose", {{"permutation", {0, 2, 1}}}),
+                     });
+}
+
+TEST_CASE(optimize_reshape_transpose_reshape_to_reshape)
+{
+    EXPECT(migraphx::optimize_shape_transforms(
+               {6, 5, 2},
+               {
+                   make_op("reshape", {{"dims", {6, 5, 2, 1}}}),
+                   make_op("transpose", {{"permutation", {0, 1, 3, 2}}}),
+                   make_op("reshape", {{"dims", {6, 10}}}),
+               }) == ops{
+                         make_op("reshape", {{"dims", {6, 10}}}),
+                     });
+}
+
+TEST_CASE(optimize_multibroadcast_transpose_reshape)
+{
+    EXPECT(migraphx::optimize_shape_transforms(
+               {1, 5, 2},
+               {
+                   make_op("multibroadcast", {{"out_lens", {20, 5, 2}}}),
+                   make_op("transpose", {{"permutation", {0, 2, 1}}}),
+                   make_op("reshape", {{"dims", {20, 10}}}),
+               }) == ops{
+                         make_op("transpose", {{"permutation", {0, 2, 1}}}),
+                         make_op("reshape", {{"dims", {1, 10}}}),
+                         make_op("multibroadcast", {{"out_lens", {20, 10}}}),
+                     });
+}
+
+TEST_CASE(optimize_resize1)
+{
+    EXPECT(migraphx::optimize_shape_transforms(
+               {3, 4, 4},
+               {
+                   make_op("reshape", {{"dims", {3, 1, 4, 1, 4}}}),
+                   make_op("multibroadcast", {{"out_lens", {3, 2, 4, 2, 4}}}),
+                   make_op("reshape", {{"dims", {3, 8, 8}}}),
+               }) == ops{
+                         make_op("unsqueeze", {{"axes", {1, 3}}}),
+                         make_op("multibroadcast", {{"out_lens", {3, 2, 4, 2, 4}}}),
+                         make_op("reshape", {{"dims", {3, 8, 8}}}),
+                     });
+}
+
+TEST_CASE(optimize_resize2)
+{
+    EXPECT(migraphx::optimize_shape_transforms(
+               {1, 1, 2, 2},
+               {
+                   make_op("reshape", {{"dims", {1, 1, 2, 1, 2, 1}}}),
+                   make_op("multibroadcast", {{"out_lens", {1, 2, 2, 2, 2, 3}}}),
+                   make_op("reshape", {{"dims", {1, 2, 4, 6}}}),
+               }) == ops{
+                         make_op("unsqueeze", {{"axes", {3, 5}}}),
+                         make_op("multibroadcast", {{"out_lens", {1, 1, 2, 2, 2, 3}}}),
+                         make_op("reshape", {{"dims", {1, 1, 4, 6}}}),
+                         make_op("multibroadcast", {{"out_lens", {1, 2, 4, 6}}}),
+                     });
+}
+
+TEST_CASE(optimize_reshape_2_squeeze)
+{
+    EXPECT(migraphx::optimize_shape_transforms({3, 1, 5, 1, 2, 1, 1},
+                                               {
+                                                   make_op("reshape", {{"dims", {3, 5, 2}}}),
+                                               }) ==
+           ops{
+               make_op("squeeze", {{"axes", {1, 3, 5, 6}}}),
+           });
```

Comment:
minor: Really poor formatting by clang-format here. Probably better to add a newline to L294